### PR TITLE
Feature table updates in preparation for styling

### PR DESCRIPTION
**Pull request by lilleyse**
A bunch of organizational changes as we get closer to implementing styling with the new model and metadata system.

* Renamed `MetadataGltfExtension` to `FetaureMetadata`
* Added `FeatureTable` which contains a `MetadataTable` and will contain additional color/show and pick textures like `Cesium3DTileBatchTable`.
* Added stubs for `FeatureTexture` and `FeatureTextureProperty`
* Removed `properties` getter from all types of metadata since we decided property access should be strictly through `getProperty`/`setProperty` functions.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @lilleyse!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

File: Source/Scene/FeatureMetadata.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+ * An object containing feature metadata.
+ * <p>
+ * See the {@link https://github.com/CesiumGS/glTF/tree/3d-tiles-next/extensions/2.0/Vendor/EXT_feature_metadata/1.0.0|EXT_feature_metadata Extension} for glTF.
+ * </p>
+ *
+ * @param {Object} options Object with the following properties:
+ * @param {String} options.extension The extension JSON object.
+ * @param {MetadataSchema} options.schema The parsed schema.
+ * @param {Object} [options.bufferViews] An object mapping bufferView IDs to Uint8Array objects.
+ * @param {Object} [options.textures] An object mapping texture IDs to {@link Texture} objects.
```
Comment (ptrgags) [inline Source/Scene/FeatureMetadata.js:17]: Check throughout the PR for this kind of doc update, I've seen it in a few places

```suggestion
 * @param {Object<String,Uint8Array>} [options.bufferViews] An object mapping bufferView IDs to Uint8Array objects.
 * @param {Object<String,Texture>} [options.textures] An object mapping texture IDs to {@link Texture} objects.
```

**Review by ptrgags — COMMENTED** [review]
@lilleyse Overall looks good, mostly minor things. Though I noticed that `MetadataEntity` doesn't have many tests even though a lot of key code happens there.

**Comment by lilleyse** [comment]
@ptrgags Updated!

**Comment by ptrgags** [comment]
Looks good, thanks @lilleyse!

**Pull request merged**